kubernetes/cache/kube_cache.go

```diff
@@ -537,6 +538,64 @@ func (c *kubeCache) GetDaemonSet(namespace, name string) (*apps_v1.DaemonSet, er
 	return retDS, nil
 }
 
+func (c *kubeCache) GetDaemonSetsWithSelector(namespace string, selectorLabels map[string]string) ([]apps_v1.DaemonSet, error) {
+	defer c.cacheLock.RUnlock()
+	c.cacheLock.RLock()
+
+	var daemonSets []apps_v1.DaemonSet
+	var err error
+	selector := labels.Set(selectorLabels)
+
+	if namespace == metav1.NamespaceAll {
+		if c.clusterScoped {
+			daemonSets, err = c.GetDaemonSets(namespace)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			for _, nsCacheLister := range c.nsCacheLister {
+				dsNS, err2 := nsCacheLister.daemonSetLister.List(labels.Everything())
```

Comment:
If you pass the label selector into the `List()` method rather than `labels.Everything()` then you can remove all of the manual filtering below. The label selector is meant to be passed into the list methods this way and the filtering will be handled by the underlying client.

See the one of the Istio methods like `GetDestinationRules` as an example: https://github.com/kiali/kiali/blob/90ceb89066c2fb82a06dd3fdcc5c875f8c8c30a7/kubernetes/cache/kube_cache.go#L869

Reply:
Yes, I don't know why when I tried something similar it was not working, that's why I added all that filtering logic. It looks to be working now